src/porepy/grids/mdg_generation.py

```diff
@@ -536,19 +536,34 @@ def _preprocess_cartesian_args(
 
     # if provided overwrite information in x-direction
     if cell_size_x is not None:
-        n_x: int = round((xmax - xmin) / cell_size_x)
+        if np.floor((xmax - xmin) / cell_size_x) < 1.0:
+            warnings.warn(
+                "In the x-direction, cell_size_x is greater than the domain. The domain "
+                "size is used instead."
+            )
+        n_x: int = int(np.max([np.floor((xmax - xmin) / cell_size_x), 1]))
```

Comment:
I think we should keep round. Otherwise, passing a slightly too large cell size will yield unexpected results. 
MVE:

```
import numpy as np
ds = 1
cs = .066667 # approx 1/15
frac = ds / cs
fl = int(np.max([np.floor(frac), 1]))
ro = int(np.max([np.round(frac), 1]))
print(frac, fl, ro)
```

Reply:
I agree with your statement. Now the code is calculating the number of cells or points using round instead of floor.